pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -503,6 +509,14 @@ func (k *KSMCheck) processMetrics(sender aggregator.Sender, metrics map[string][
 		for _, metricFamily := range metricsList {
 			// First check for aggregator, because the check use _labels metrics to aggregate values.
 			if aggregator, found := k.metricAggregators[metricFamily.Name]; found {
+				if metricFamily.Name == "kube_configmap_info" && !k.instance.EnableConfigMapCount {
+					log.Tracef("ConfigMap metric found but counting ConfigMaps is disabled.")
+					continue
+				}
+				if metricFamily.Name == "kube_secret_info" && !k.instance.EnableSecretCount {
+					log.Tracef("Secret metric found but counting Secrets is disabled.")
+					continue
+				}
```

Comment:
imo, this is not needed if we remove the 2 options as I suggested in a previous comment.